src/pyqasm/pulse/openpulse_visitor.py

```diff
@@ -0,0 +1,379 @@
+                            print("ERROR")
+                        duration_arg = wave_form_args[1]
+                        if isinstance(duration_arg, pulse_ast.DurationType):
+                            print("TODO")
+                        elif isinstance(duration_arg, pulse_ast.Identifier):
+                            print("TODO")
+                        else:
+                            print("ERROR")
+                        std_deviation_arg = wave_form_args[2]
+                        if isinstance(std_deviation_arg, pulse_ast.DurationType):
+                            print("TODO")
+                        elif isinstance(std_deviation_arg, pulse_ast.Identifier):
+                            print("TODO")
+                        else:
+                            print("ERROR")
+                    else:
+                        print("ERROR")
+
+                if id_func_name == "sech":
+                    print("TODO")
+                if id_func_name == "gaussian_square":
+                    print("TODO")
+                if id_func_name == "drag":
+                    print("TODO")
+                if id_func_name == "constant":
+                    print("TODO")
+                if id_func_name == "sine":
+                    print("TODO")
+                if id_func_name == "mix":
+                    print("TODO")
+                if id_func_name == "sum":
+                    print("TODO")
+                if id_func_name == "phase_shift":
+                    print("TODO")
+                if id_func_name == "scale":
+                    print("TODO")
+
+                ## Add all waveforms to global scope
+
+        if self._check_only:
+            return []
+
+        return []
+
+    def _visit_function_call(
+        self, statement: pulse_ast.FunctionCall
+    ) -> tuple[Any, list[pulse_ast.Statement]]:
+        """Visit a function call element.
+
+        Args:
+            statement (pulse_ast.FunctionCall): The function call to visit.
+        Returns:
+            None
+
+        """
+        # evaluate expressions to get name
+        if statement.name.name == "set_phase":
+            stmt_args = statement.arguments
+            if len(stmt_args) == 2:
+                frame_arg = stmt_args[0]
+                if isinstance(frame_arg, pulse_ast.Identifier):
+                    print("TODO")
+                    # get frame object from global scope
+                else:
+                    print("ERROR")
+
+                angle_arg = stmt_args[1]
+                if isinstance(angle_arg, pulse_ast.Identifier):
+                    print("TODO")
+                    # evaluate expression to get value
+                elif isinstance(angle_arg, pulse_ast.AngleType):
+                    print("TODO")
+                else:
+                    print("ERROR")
+
+            # update frame.phase
+
+            else:
+                print("ERROR")
+        if statement.name.name == "get_phase":
```

Comment:
I think we should refactor all the pulse specific keywords like `get_phase`, `shift_phase`, etc. as `Enum` or equivalent inside the  `pulse/maps.py`. It would be cleaner to identify these with specific naming conventions such as `PulseMethod.<name>`, `PulseWaveform.<name>`, etc. and refer here